app/controllers/provider/sessions_controller.rb

```diff
@@ -34,6 +34,8 @@ def create
     else
       new
       flash.now[:error] ||= strategy&.error_message
+      attempted_cred = auth_params.fetch(:username, nil) || sso_params.fetch(:token, nil)
+      AuditLogService.call("Login attempt failed: #{domain_account.external_admin_domain} - #{attempted_cred}")
```

Comment:
is token safe to log? Can it be a valid token for something else so putting in logs to be inappropriate? What do we already have in `strategy&.error_message`? I wonder if we can take example from it.

Reply:
I didn't think on this. Maybe we should not log the token. I'd say those tokens are one timers with a short expiration time. But not 100% sure so better remove it, I think. We can log the attempted username if any, if a SSO login failed, then we don't log the user because we don't have it.

Reply:
Done here: https://github.com/3scale/porta/pull/3885/commits/38d22eada58fcf5802a585724ac15872c82df170

Reply:
Maybe log this `strategy&.error_message` appended to the existing one. Also will be very useful to log the client IP address I believe. Will be helpful before resorting to traffic capture.

Reply:
Yeah I can do that 👍

Reply:
Done: https://github.com/3scale/porta/pull/3885/commits/22ce1889708acf065aa65f3bf90999d68794e218